osf/utils/migrations.py

```diff
@@ -390,6 +390,30 @@ def create_schema_blocks_for_question(state, rs, question, sub=False):
         # If there are multiple choice answers, create blocks for these as well.
         split_options_into_blocks(state, rs, question, schema_block_group_key)
 
+def create_schema_blocks_for_atomic_schema(schema):
+    """
+    Atomic schemas are a short cut around making an typical metaschemas by being totally explict about the schemablocks
+    being created.
+    """
+
+    from osf.models import RegistrationSchemaBlock
+    schema_block_group_key = generate_object_id()
+    for index, block in enumerate(schema.schema['blocks']):
+        block_type = block['block_type']
+
+        if block_type == 'question-label':
+            schema_block_group_key = generate_object_id()
+        elif block_type in RegistrationSchemaBlock.INPUT_BLOCK_TYPES:
+            registration_response_key = f'{schema.id}-{index}'
+        else:
+            registration_response_key = None
```

Comment:
this is inheriting the `schema_block_group_key` from the most recent question when it shouldn't.

Reply:
Well yeah but as far as I can see, it doesn't actually cause problems, also fun fact two text input boxes with the same group key become completely synced, each mirroring each other key stroke for key stroke.